File: docs/snippets/python/cache/disk_cache.py
```python title="disk_cache.py"
from pathlib import Path
from xberg import ExtractInput, Xberg, ExtractionConfig, CacheConfig

cache_dir = Path.home() / ".cache" / "xberg"
cache_dir.mkdir(parents=True, exist_ok=True)

config = ExtractionConfig(
    use_cache=True,
    cache_config=CacheConfig(
        cache_path=str(cache_dir),
        max_cache_size=500 * 1024 * 1024,
        cache_ttl_seconds=7 * 86400,
        enable_compression=True,
    ),
)

xberg = Xberg(config)

print("First extraction (will be cached)...")
result1 = xberg.extract(ExtractInput.from_uri("document.pdf"), ExtractionConfig())
print(f"  - Content length: {len(result1.results[0].content)}")
print(f"  - Cached: {result1.results[0].metadata.get('was_cached', False)}")

print("\nSecond extraction (from cache)...")
result2 = xberg.extract(ExtractInput.from_uri("document.pdf"), ExtractionConfig())
print(f"  - Content length: {len(result2.results[0].content)}")
print(f"  - Cached: {result2.results[0].metadata.get('was_cached', False)}")

print(f"\nResults are identical: {result1.results[0].content == result2.results[0].content}")

cache_stats = xberg.get_cache_stats()
print(f"\nCache Statistics:")
print(f"  - Total entries: {cache_stats.get('total_entries', 0)}")
print(f"  - Cache size: {cache_stats.get('cache_size_bytes', 0) / 1024 / 1024:.1f} MB")
print(f"  - Hit rate: {cache_stats.get('hit_rate', 0):.1%}")
```

File: docs/snippets/ruby/cache/disk_cache.rb
```ruby title="disk_cache.rb"
require 'xberg'
require 'fileutils'

cache_dir = File.expand_path('~/.cache/xberg')
FileUtils.mkdir_p(cache_dir)

config = Xberg::ExtractionConfig.new(
  use_cache: true,
  cache_config: Xberg::CacheConfig.new(
    cache_path: cache_dir,
    max_cache_size: 500 * 1024 * 1024,
    cache_ttl_seconds: 7 * 86400,
    enable_compression: true,
  )
)

puts "First extraction (will be cached)..."
input1 = Xberg::ExtractInput.new(uri: 'document.pdf')
result1 = Xberg.extract(input1, config)
doc1 = result1.results.first
puts "  - Content length: #{doc1.content.length}"
puts "  - Cached: #{doc1.metadata['was_cached']}"

puts "\nSecond extraction (from cache)..."
input2 = Xberg::ExtractInput.new(uri: 'document.pdf')
result2 = Xberg.extract(input2, config)
doc2 = result2.results.first
puts "  - Content length: #{doc2.content.length}"
puts "  - Cached: #{doc2.metadata['was_cached']}"

puts "\nResults are identical: #{doc1.content == doc2.content}"

cache_stats = Xberg.get_cache_stats
puts "\nCache Statistics:"
puts "  - Total entries: #{cache_stats['total_entries']}"
puts "  - Cache size: #{(cache_stats['cache_size_bytes'] / 1024.0 / 1024.0).round(1)} MB"
puts "  - Hit rate: #{(cache_stats['hit_rate'] * 100).round(1)}%"
```
